lib/src/kvs/ds.rs

```diff
@@ -381,7 +402,15 @@ impl Datastore {
 					}
 				}
 			}
+		};
+		if resolve_err.is_err() {
+			err.push(resolve_err.unwrap_err());
+		}
+		if err.is_empty() {
+			error!("Error bootstrapping sweep phase: {:?}", err);
+			return Err(Error::Tx(format!("Error bootstrapping sweep phase: {:?}", err)));
```

Comment:
Is it still okay for this function to return any error to the caller?

I mean, I thought we had something like `dbs.bootstrap().await?;` that prevents the node init process from proceeding on any error.

Should we also modify the call-side, or let this function not return any error, or...?

Reply:
The think about bootstrapping, is that there are soft errors (recoverable, such as cleaning up a partially corrupted live query state) and hard errors (something unimaginable has happened like the storage engine rejecting scans). 

We want to complete the bootstrap as much as possible, but we don't want to lose the errors I think.

The reason for keeping the errors is because we may want to retry the transaction, but still have metrics for error types. (ex. retry 5 times, if by the 5th time still errors then register metric).

Reply:
Marking as resolved, but definitely reverse that if you think otherwise @mumoshu . There are arguments on both sides and I am not 100% convinced about the interface.
By surfacing errors always, it makes the bootstrap easier to test (side effects + errors as opposed to side effects + maybe capturing logs).

Reply:
Thanks @phughk! I do love the idea of distinguishing soft errors vs. hard errors.
The problem I think is then, there's no comment in this function (or elsewhere) to say that this function is supposed to return hard errors only and soft errors are supposed to be retried within this function.
If that makes sense, can we perhaps add some comment to denote that?

If I missed anything and made a wrong feedback, just reverse this thread once again!

Reply:
Added comment to bootstrap explaining it has side effects alongside errors 👍